### PR TITLE
feat(DTFS2-7601): add e2e coverage for cancelled bss ewcs in portal

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
When a deal is cancelled in tfm, the statuses should be updated in portal. We want test coverage for this

## Resolution :heavy_check_mark:
- Create two new tests for BSS/EWCS AIN and MIN deals, checking the status is `Cancelled` and the facility status `risk expired`
- Create helper to cancel deal in tfm

### CODE REVIEW

File: e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import portalPages from '../../../../../../../portal/cypress/e2e/pages';
+import MOCK_USERS from '../../../../../../../e2e-fixtures/portal-users.fixture';
+import generateAinReadyToSubmit from '../../test-data/AIN-deal/dealReadyToSubmit';
+import { TFM_URL, PIM_USER_1 } from '../../../../../../../e2e-fixtures';
+import { yesterday } from '../../../../../../../e2e-fixtures/dateConstants';
+import { submitDealCancellation } from '../../../../../support/trade-finance-manager-ui/submit-deal-cancellation';
+
+const { BANK1_MAKER1, BANK1_CHECKER1 } = MOCK_USERS;
+
+describe('Deal Cancellation', () => {
```
Comment (ttbarnes) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js:10]: Can we update the file name and the test description to include "-status-updates" or similar

File: e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      portalPages.contract.checkDealDetailsTab().click();
+
+      cy.assertText(portalPages.contract.status(), 'Cancelled');
+    });
+
+    it('displays facility status `Risk expired` on deal summary page', () => {
+      portalPages.contract.visit(deal);
+
+      cy.assertText(portalPages.contract.bondTransactionsTable.row(dealFacilities[1]._id).bondStatus(), 'Risk expired');
+      cy.assertText(portalPages.contract.loansTransactionsTable.row(dealFacilities[0]._id).loanStatus(), 'Risk expired');
```
Comment (ttbarnes) [inline e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js:82]: Constants?

File: e2e-tests/ukef/cypress/support/trade-finance-manager-ui/submit-deal-cancellation.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { TFM_URL } from '../../../../e2e-fixtures';
+import * as caseDealPage from '../../../../tfm/cypress/e2e/pages/caseDealPage';
+import * as checkDetailsPage from '../../../../tfm/cypress/e2e/pages/deal-cancellation/check-details';
+
+export const submitDealCancellation = ({ dealId, effectiveDate = undefined }) => {
```
Comment (ttbarnes) [inline e2e-tests/ukef/cypress/support/trade-finance-manager-ui/submit-deal-cancellation.js:5]: Can we add some documentation please

Comment (ttbarnes) [inline e2e-tests/ukef/cypress/support/trade-finance-manager-ui/submit-deal-cancellation.js:5]: I don't think we need the `effectiveDate` to have a default of undefined - would be undefined if it's not provided 😛

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    });
+
+    it('displays deal status `Cancelled` on deal summary page', () => {
+      portalPages.contract.visit(deal);
+
+      cy.assertText(portalPages.contract.status(), 'Cancelled');
+
+      portalPages.contract.checkDealDetailsTab().click();
+
+      cy.assertText(portalPages.contract.status(), 'Cancelled');
```
[the remark this answers is not in the document]
Comment (MarRobSoftwire) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js]: This is taken from the ACs https://ukef-dtfs.atlassian.net/browse/DTFS2-7306
> When I select either ‘View’ or ‘Check deal details’ tab

File: e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      portalPages.contract.proceedToSubmit().click();
+
+      portalPages.contractConfirmSubmission.confirmSubmit().check();
+      portalPages.contractConfirmSubmission.acceptAndSubmit().click(deal);
+
+      cy.clearCookie('dtfs-session');
+      cy.clearCookie('_csrf');
+      cy.getCookies().should('be.empty');
+      cy.forceVisit(TFM_URL);
+      cy.tfmLogin(PIM_USER_1);
```
[the remark this answers is not in the document]
Comment (MarRobSoftwire) [reply e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/BSS-EWCS/AIN-cancellation-updates-portal.spec.js:63]: This is the UKEF tests directory, as far as I know these are all tests that do stuff in both TFM and portal. Those in this `portal` directory seem to be checking portal is updated after changes have been made in tfm

**Review by ttbarnes — APPROVED** [review]
Nice work

**Pull request merged**